src/main/java/roboy/util/TelegramCommunicationHandler.java

```diff
@@ -0,0 +1,339 @@
+            }
+            else {
+                // sum all of the messages
+                String oldMessage = result.values().toArray()[0].toString();
+                String newMessage = oldMessage + " " + message;
+
+                //equal chat id
+                result = new HashMap<>();
+                result.put(chatID, newMessage);
+            }
+        }
+
+
+        if(result != null) {
+            // notify the input device
+            pairs.removeAll(removedObjects);
+            TelegramInput.onUpdate(result);
+        }
+    }
+
+    // ---- SEND METHODS ----
+
+    /**
+     * Called from the OutputDevice when a message desired to send
+     * Initiates the "typing status" and waits for a specified time
+     * Sends the message afterwards
+     * @param chatID unique identifier for a chat.
+     */
+    public void sendMessage(String message, String chatID){
+        try {
+            sendTypingFromChatID(chatID);
+
+            TimeUnit.SECONDS.sleep(TYPING_TIME_LIMIT);
+
+            SendMessage sendMessageRequest = new SendMessage();
+            sendMessageRequest.setChatId(chatID);//who should get the message? the sender from which we got the message...
+            sendMessageRequest.setText(message);
+            try {
+                execute(sendMessageRequest);
+            } catch (TelegramApiException e) {
+                //do some error handling
+            }//end catch()
+
+        } catch (InterruptedException e) {
+            e.printStackTrace();
+        }
+    }
+
+    /**
+     * Called from the OutputDevice when a message desired to send with stickers
+     * Directly sends the sticker, without waiting for a specified time!
+     * @param chatID unique identifier for a chat.
+     * @param stickerId unique identifier for a sticker.
+     */
+    public void sendSticker(String chatID, String stickerId){
+        SendSticker sendStickerRequest = new SendSticker();
+        try {
+            sendStickerRequest.setChatId(chatID);
+            sendStickerRequest.setSticker(stickerId);
+
+            sendSticker(sendStickerRequest);
+        } catch (TelegramApiException e) {
+            e.printStackTrace();
+        }catch (Exception e){
+            e.printStackTrace();
+        }
+    }
+
+
+    /**
+     * Sends the "typing" status to the telegram chat
+     * @param chatID unique identifier for a chat.
+     */
+    public void sendTypingFromChatID(String chatID){
+        try {
+            SendChatAction action = new SendChatAction();
+            action.setChatId(chatID);
+            action.setAction(ActionType.TYPING);
+            execute(action);
+        } catch (TelegramApiException e) {
+            e.printStackTrace();
+        }
+    }
+
+    /**
+     * Sends a image from url to the desired chat
+     * @param url image's url
+     * @param chatId unique identifier for a chat
+     */
+    public void sendImageFromUrl(String url, String chatId) {
```

Comment:
This is interesting!
Maybe we should have an ImageAction besides SpeechAction and EmotionAction. Not in this PR of course ;)